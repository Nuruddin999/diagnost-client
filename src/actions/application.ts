import { createAction } from "@reduxjs/toolkit";
export const Types = {
  applicationAdd: 'application/add',
  applicationGet: 'application/get',
  applicationGetOne: 'application/getone',
  applicationUpdate: 'application/update',
  applicationDel: 'application/deleteone',
  applicationChangeDeleteOption: 'application/changedelopt',
}
export type applicationForAdd = {
  id?: number,
  creator?: string,
  patientName: string,
  patientBirthDate: string,
  patientRequest: string,
  fundName: string,
  fundRequest: string,
  manager: string,
  managerSpeciality: string,
  managerId: string,
  creationDate: string,
  execDate: string,
}

export const addApplication = createAction(Types.applicationAdd, function prepare(application: applicationForAdd) {
  return {
    payload: {
      ...application
    },
  }
})
export const getApplication = createAction(Types.applicationGet, function prepare(page: number,
  limit: number,
  manager: string,
  patientName: string,
  patientRequest: string,
  fundName: string,
  fundRequest: string,
  creator: string) {
  return {
    payload: {
      page, limit, manager, patientName, patientRequest, fundName, fundRequest, creator
    },
  }
})
export const changeDeleteOptionAction = createAction(Types.applicationChangeDeleteOption, function prepare(id: string) {
  return {
    payload: {
      id
    },
  }
})

export const getOneApplication = createAction(Types.applicationGetOne, function prepare(id: string) {
  return {
    payload: {
      id
    },
  }
})

export const deleteOneApplication = createAction(Types.applicationDel, function prepare(id: string) {
  return {
    payload: {
      id
    },
  }
})
/**
 * экш обновления заключения
 */
export const updateApplication = createAction(Types.applicationUpdate)

