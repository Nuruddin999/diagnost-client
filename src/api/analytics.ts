import {diagnostApi} from "./index";
import {UserItemRecapType} from "../common/types";



export const getUsersRecapApi = async (period: string, id:string,  fromD:string,toD:string
): Promise<{users:Array<UserItemRecapType>, count:number}> => {
    const response = await diagnostApi.get('/gurec', {
        headers: {'Authorization': `Bearer ${localStorage.getItem('refreshToken')}`},
        params: {period, fromD, toD},
    })
    return response.data
}

export const getUserItemRecapApi = async (period: string, id:string, fromD?: string, toD?:string
): Promise<UserItemRecapType> => {
    const response = await diagnostApi.get('/gutrec', {
        headers: {'Authorization': `Bearer ${localStorage.getItem('refreshToken')}`},
        params: {period, id, fromD,toD,},
    })
    return response.data
}


export const getUserItemRecapApiByPeriod = async (fromD: string, toD:string,id:string
): Promise<UserItemRecapType> => {
    const response = await diagnostApi.get('/gutrecbpr', {
        headers: {'Authorization': `Bearer ${localStorage.getItem('refreshToken')}`},
        params: {fromD,toD, id},
    })
    return response.data
}