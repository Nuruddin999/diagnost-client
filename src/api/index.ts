import axios from "axios";

export const diagnostApi = axios.create({
    baseURL: 'http://188.68.222.107/api/',
    headers: {'Authorization': `Bearer ${localStorage.getItem('refreshToken')}`}
  });