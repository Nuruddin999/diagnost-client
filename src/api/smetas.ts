import {diagnostApi} from "./index";
import {SmetasResponseType} from "../common/types";

/**
 * Получаем список заключений.
 * @param {number} page Страница.
 * @param {number} limit Сколько показывать.
 * @returns
 */
export const getSmetasApi = async (page: number,
                                        limit: number):Promise<SmetasResponseType> => {
    const response = await diagnostApi.get('/smetas', {
        headers: { 'Authorization': `Bearer ${localStorage.getItem('refreshToken')}` },
        params: { page,limit }
    })
    return response.data
}