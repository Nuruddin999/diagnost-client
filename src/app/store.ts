import { Action, combineReducers, configureStore, ThunkAction } from '@reduxjs/toolkit';
import createSagaMiddleware from 'redux-saga';
import applicationSlice from '../reducers/applicationSlice';
import userSlice from '../reducers/userSlice';
import runSagas from '../sagas';

const rootReducer = combineReducers({
  user: userSlice,
  application:applicationSlice
})

export type RootState = ReturnType<typeof rootReducer>

const sagaMiddleware = createSagaMiddleware()

export const store = configureStore({
  reducer: rootReducer,
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware().concat(sagaMiddleware),
});

sagaMiddleware.run(runSagas)

export type AppDispatch = typeof store.dispatch;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;