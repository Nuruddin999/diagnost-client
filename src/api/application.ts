import { applicationForAdd } from '../actions/application';
import { applicationInitialState } from '../reducers/applicationItemSlice';
import { diagnostApi } from './index';
export const addApplicationApi = async (application: applicationForAdd) => {
    const response = await diagnostApi.post('/application', {
        ...application
    })
    return response.data
}
/**
 * Получаем список заключений.
 * @param {number} page Страница.
 * @param {number} limit Сколько показывать.
 * @returns
 */
export const getApplicationApi = async (page: number, limit: number) => {
    const response = await diagnostApi.get('/applications', {
        params: { page, limit }
    })
    return response.data
}
/**
 * Получение одного заключения по id.
 * @param {number} id Id заключения.
 */
export const getOneApplicationApi = async (id: string) => {
    const response = await diagnostApi.get(`/applications/${id}`)
    return response.data
}
/**
 * Запрос обновления одного заключения.
 * @param {Object} application обновленное заключение.
 */
export const updateOneApplicationApi = async (application: applicationInitialState) => {
    ('application in api', application)
    const response = await diagnostApi.post('/updappl', {
        ...application
    })
    return response.data
}


