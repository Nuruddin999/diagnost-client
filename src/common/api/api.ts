import { diagnostApi } from "../../api"

/**
 * Получение одного элемента по id.
 * @param {number} id Id заключения.
 *  @param {string} url Id заключения.
 */
export const getListItemById = async (id: string, url: string) => {
    const response = await diagnostApi.get(`/${url}/${id}`, {
        headers: { 'Authorization': `Bearer ${localStorage.getItem('refreshToken')}` },
    })
    return response.data
}
export const sessionHeartBit = async (url:string, id: number | string, duration:number) => {
    const response = await diagnostApi.post(url, {id, duration}, {
        headers: {
            'Authorization': `Bearer ${localStorage.getItem('refreshToken')}`,
        },
    })
    return response.data
}