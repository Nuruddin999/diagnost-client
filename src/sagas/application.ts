
import { getListItemById } from './../common/api/api';
import { changeIsDeletedPlace, deleteOneApplicationApi, getApplicationApi, updateOneApplicationApi } from './../api/application';
import { applicationForAdd, getApplication, Types } from './../actions/application';
import { call, put, select, all, race } from "redux-saga/effects"
import { addApplicationApi } from '../api/application';
import { saveApplicationsList } from '../reducers/applicationSlice';
import { applicationInitialState, saveApplicationItem, saveCheckupPlanDeletedPlace, successUpdate } from '../reducers/applicationItemSlice';
import { RootState } from '../app/store';
import { setCircular, setError, setStatus } from '../reducers/ui';
type applicationAddResponse = {
  id: number,
  patientName: string,
  patientBirthDate: string,
  patientRequest: string,
  fundName: string,
  fundRequest: string,
  manager: string,
  managerSpeciality: string,
  managerId: string,
  checkUpPlaceIsDeleted: boolean,
  creationDate: string,
  execDate: string,
  updatedAt: string,
  createdAt: string
}
type getAllApplicationsResponse = {
  count: number,
  rows: Array<applicationAddResponse>,
}
export type consiliumDoctor = {
  id?: number,
  name: string,
  speciality: string,
}

type applicationItemFields = {
  ConsiliumDoctors: Array<consiliumDoctor>,
  Diagnostics: Array<{
    id?: number,
    diagnosis: string,
  }>,
  mostProblDiagnosis: string,
  secondaryDiagnosis: string,
  anamnesis: string,
  complaint: string,
  patientName: string,
  managerSignUrlPath: string,
  diagnosticData: string,
  patientBirthDate: string,
  CheckupPlans: Array<{
    id?: number,
    kind?: string,
    place?: string,
    target?: string
  }>
  Comments: Array<{
    title?: string,
    comment: string,
  }>,
}
export type applicationItemResponse = applicationAddResponse & applicationItemFields;
/**
 * Вход в систему.
 * @param login .
 */

export function* addApplication(addApplication: { type: 'application/add', payload: applicationForAdd }) {
  try {
    yield put(setStatus('pending'))
    const { id: userId, role } = yield select((state: RootState) => state.user.user)
    const response: getAllApplicationsResponse = yield call(addApplicationApi, addApplication.payload)
    if (response) {
      yield put(setStatus('success'))
      yield put(getApplication(1, 10, '', '', '', '', '', role === 'doctor' ? userId : 'all'))
    }
  } catch (e: any) {
    if (e.response) {
    }
    else {
    }
  }
}
/**
 * Сага получения списка заключений.
 * @param addApplication
 */
export function* fetchApplication(getApplication: { type: 'application/get', payload: { page: number, limit: number, manager: string, patientName: string, patientRequest: string, fundName: string, fundRequest: string, creator: string } }) {
  try {
    yield put(setStatus('pending'))
    const { page, limit, patientName, patientRequest, fundName, fundRequest, manager, creator } = getApplication.payload
    const response: getAllApplicationsResponse = yield call(getApplicationApi, page, creator, limit, manager, patientName, patientRequest, fundName, fundRequest)
    if (response) {
      const { rows, count } = response
      yield put(saveApplicationsList({ applications: rows, count }))
      yield put(setStatus('ok'))
    }
  } catch (e: any) {
    yield all([put(setStatus('no')), put(setError('Произошлав ошибка, повторите попозже'))])
  }
}
/**
 * Получение заключения по Id.
 * @param {Object} getApplication .
 */
export function* fetchOneApplication(getApplication: { type: 'application/getone', payload: { id: string } }) {
  try {
    const { id } = getApplication.payload
    const response: applicationItemResponse = yield call(getListItemById, id, 'applications')
    if (response) {
      yield put(saveApplicationItem({ ...response }))
    }
  } catch (e: any) {
    if (e.response) {
    }
    else {
    }
  }
}
/**
 * Обновление  заключения.
 * @param {Object} getApplication .
 */
export function* updateOneApplication(updateApplication: { type: 'application/update' }) {

  try {
    const application: applicationInitialState = yield select((state: RootState) => state.applicationItem)
    const consiliumDoctorsFiltered = application.consiliumDoctors.map(doctor => ({ name: doctor.name, speciality: doctor.speciality }))
    yield put(setCircular(true))
    const response: applicationItemResponse = yield call(updateOneApplicationApi, { ...application, consiliumDoctors: consiliumDoctorsFiltered, execDate: new Date().toString() })
    if (response) {
      yield put(setCircular(false))
      yield put(successUpdate('success'))
    }
  } catch (e: any) {
    if (e.response) {
      yield put(setCircular(false))
    }
    else {
      yield put(setCircular(false))
    }
  }
}

/**
 * Изменение опции удаления места в плане лечения.
 */
export function* changeDeleteOptionInPlan(body: {type: 'application/changedelopt', payload :{ id: string}}) {
  try {
    const response: { checkUpPlaceIsDeleted: boolean } = yield call(changeIsDeletedPlace, body.payload.id)
    if (response) {
      yield put(setCircular(false))
      yield put(saveCheckupPlanDeletedPlace(response.checkUpPlaceIsDeleted))
      yield put(successUpdate('success'))
    }
  } catch (e: any) {
    if (e.response) {
      yield put(setCircular(false))
    }
    else {
      yield put(setCircular(false))
    }
  }
}



/**
* Удаление  заключения.
* @param {Object} getApplication .
*/
export function* removeOneApplication(delApplication: { type: 'application/deleteone', payload: { id: string } }) {

  try {
    const { id } = delApplication.payload
    const { id: userId, role } = yield select((state: RootState) => state.user.user)
    const response: {} = yield call(deleteOneApplicationApi, id)
    if (response) {
      yield put(successUpdate('success'))
      yield put(getApplication(1, 10, '', '', '', '', '', role === 'doctor' ? userId : 'all'))
    }
  } catch (e: any) {
    if (e.response) {
    }
    else {
    }
  }
}