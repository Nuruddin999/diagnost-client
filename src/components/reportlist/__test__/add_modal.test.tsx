import React from 'react'
import { render, fireEvent, screen, getByPlaceholderText, getByLabelText } from '@testing-library/react';
import { Provider } from 'react-redux';
import { rootReducer, store } from '../../../app/store';
import {
  BrowserRouter as Router,
} from "react-router-dom";
import AddModal from '../add_modal';
import { createStore } from '@reduxjs/toolkit';
import { addApplicationApi  } from '../../../api/application';

jest.mock('../../../api/application',)

describe('add modal works properly', () => {
  beforeAll(() => {
    // add window.matchMedia
    // this is necessary for the date picker to be rendered in desktop mode.
    // if this is not provided, the mobile mode is rendered, which might lead to unexpected behavior
    Object.defineProperty(window, "matchMedia", {
      writable: true,
      value: (query: string) => ({
        media: query,
        // this is the media query that @material-ui/pickers uses to determine if a device is a desktop device
        matches: query === "(pointer: fine)",
        onchange: () => {},
        addEventListener: () => {},
        removeEventListener: () => {},
        addListener: () => {},
        removeListener: () => {},
        dispatchEvent: () => false,
      }),
    });
  });

  afterAll(() => {
    // @ts-ignore
    delete window['matchMedia'];
  })
  const renderAuth = () => (
    <Provider store={createStore(rootReducer, {})}>
      <Router>
        <AddModal />
      </Router>
    </Provider>
  )
  it('renders properly', () => {
    const { container } = render(renderAuth())
    expect(container).toMatchSnapshot()
  })
  it('creates new request by doctor correctly', () => {
    (addApplicationApi as unknown as jest.Mock).mockResolvedValue({
      status:'ok'
    })
    const { getByPlaceholderText, getByTestId } = render(renderAuth())
    const fioTextField = getByPlaceholderText('ФИО пациента') as HTMLInputElement;
    expect(fioTextField).not.toBeNull()
    fireEvent.change(fioTextField, { target: { value: 'Aliev Ali Alievich' } })
    expect(fioTextField.value).toBe('Aliev Ali Alievich')
    const startDateInput = getByPlaceholderText('dd.mm.y')
    fireEvent.change(startDateInput, { target: { value: "19.06.1987" } });
    expect((startDateInput as HTMLInputElement).value).toBe("19.06.1987");
    const patientRequest = getByPlaceholderText('Запрос пациента') as HTMLInputElement;
    expect(patientRequest).not.toBeNull()
    fireEvent.change(patientRequest, { target: { value: 'Нужны лекарства от астмы' } })
    expect(patientRequest.value).toBe('Нужны лекарства от астмы')
    const fundName = getByPlaceholderText('Название фонда') as HTMLInputElement;
    expect(fundName.value).toBe('')
    expect(fundName).not.toBeNull()
    fireEvent.change(fundName, { target: { value: 'Закят' } })
    expect(fundName.value).toBe('Закят')
    const fundRequest = getByPlaceholderText('Запрос фонда') as HTMLInputElement;
    expect(fundRequest.value).toBe('')
    expect(fundRequest).not.toBeNull()
    fireEvent.change(fundRequest, { target: { value: 'Нужны лекарства' } })
    expect(fundRequest.value).toBe('Нужны лекарства')
    expect(screen.queryByTestId('manager-filed')).not.toBeInTheDocument()
    fireEvent.submit(getByTestId('addrequest-form'))
    expect(fundRequest.value).toBe('')
    expect(fundName.value).toBe('')
    expect(patientRequest.value).toBe('')
    expect(fioTextField.value).toBe('')
  })
})