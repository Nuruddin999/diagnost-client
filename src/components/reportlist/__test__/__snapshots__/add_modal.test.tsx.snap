// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`add modal works properly renders properly 1`] = `
<div>
  <div
    class="add-modal-container"
  >
    <div
      class="add-form-wrapper"
    >
      <form
        data-testid="addrequest-form"
      >
        <div
          class="close-ic"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="CloseIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
            />
          </svg>
        </div>
        <div
          class="form"
        >
          <div
            class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                id="mui-1"
                placeholder="ФИО пациента"
                required=""
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-hdw1oc"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div>
            <div
              class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeSmall MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="true"
                for="mui-7"
                id="mui-7-label"
              >
                Дата рождения
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall MuiInputBase-inputAdornedEnd css-19qh8xo-MuiInputBase-input-MuiOutlinedInput-input"
                  id="mui-7"
                  placeholder="dd.mm.y"
                  type="tel"
                  value="26.12.2022"
                />
                <div
                  class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeSmall css-1laqsz7-MuiInputAdornment-root"
                >
                  <button
                    aria-label="Choose date, selected date is 26 дек. 2022 г."
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                      data-testid="CalendarIcon"
                      focusable="false"
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-1in441m"
                  >
                    <span>
                      Дата рождения
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                id="mui-4"
                placeholder="Запрос пациента"
                required=""
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-hdw1oc"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                id="mui-5"
                placeholder="Название фонда"
                required=""
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-hdw1oc"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
          <div
            class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
          >
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-sizeSmall css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
            >
              <input
                aria-invalid="false"
                class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputSizeSmall css-1n4twyu-MuiInputBase-input-MuiOutlinedInput-input"
                id="mui-6"
                placeholder="Запрос фонда"
                required=""
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-hdw1oc"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <button
          class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButtonBase-root add-button css-11qr2p8-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="submit"
        >
          <div
            class="btn-no-loader"
            data-testid="loaded"
          >
            <span
              class="btn-title"
            >
              Добавить задание
            </span>
          </div>
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </form>
    </div>
  </div>
</div>
`;
